modules/internal/localeModels/flat/LocaleModel.chpl

```diff
@@ -286,32 +286,44 @@ module LocaleModel {
   private extern proc chpl_memhook_md_num(): chpl_mem_descInt_t;
 
   // The allocator pragma is used by scalar replacement.
+
+  // Note that there's 2 chpl_here_alloc() procs. This one explicitly takes an
+  // int(64) and is marked with "locale model alloc" while the one below takes
+  // a generic `integral`. This one is used/called by the compiler and so its
+  // type must be fully defined.
   pragma "allocator"
   pragma "locale model alloc"
```

Comment:
I don't see any documentation for pragma "locale model alloc". Based on a glance at the compiler sources I'm guessing each locale model has to have some function whose prototype is like this one as being the thing to call to allocate memory 'here', but I suspect there's more to it than that.  (With hierarchical locale models do we need one in every LocModel module, or must there be precisely one in the compilation? Whatever the answer is, does anything enforce this, or do we silently go astray if the rule is broken? Etc.) Can you add to the comment to give enough guidance to future maintainers to know what to do (or not do) with this?